crates/rolldown/src/ecmascript/ecma_generator.rs

```diff
@@ -21,9 +21,7 @@ pub struct EcmaGenerator;
 
 impl Generator for EcmaGenerator {
   #[allow(clippy::too_many_lines)]
-  async fn render_preliminary_assets<'a>(
-    ctx: &GenerateContext<'a>,
-  ) -> Result<Vec<PreliminaryAsset>> {
+  async fn render_preliminary_assets<'a>(ctx: &mut GenerateContext<'a>) -> Result<GenerateOutput> {
```

Comment:
```suggestion
  async fn render_preliminary_assets<'a>(ctx: &mut GenerateContext<'a>) -> Result<DiagnosableResult<GenerateOutput>> {
```

`.errors:` is not the recommended pattern anymore. It could not break the control flow.

Reply:
Good idea. Let me update it.